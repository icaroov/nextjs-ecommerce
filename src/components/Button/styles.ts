import styled, { css, DefaultTheme } from 'styled-components'
import { darken } from 'polished'

import { ButtonProps } from '.'

type ContainerProps = {
  hasIcon: boolean
} & Pick<ButtonProps, 'size' | 'fullWidth' | 'minimal'>

const containerModifiers = {
  small: (theme: DefaultTheme) => css`
    height: 3rem;
    font-size: ${theme.font.sizes.xsmall};
  `,
  medium: (theme: DefaultTheme) => css`
    height: 4rem;
    font-size: ${theme.font.sizes.small};
    padding: ${theme.spacings.xxsmall} ${theme.spacings.medium};
  `,
  large: (theme: DefaultTheme) => css`
    height: 5rem;
    font-size: ${theme.font.sizes.medium};
    padding: ${theme.spacings.xsmall} ${theme.spacings.xlarge};
  `,
  fullWidth: () => css`
    width: 100%;
  `,
  withIcon: (theme: DefaultTheme) => css`
    svg {
      width: 1.5rem;

      & + span {
        margin-left: ${theme.spacings.xxsmall};
      }
    }
  `,
  minimal: (theme: DefaultTheme) => css`
    background: none;
    color: ${theme.colors.primary};

    &:hover {
      color: ${darken(0.2, theme.colors.primary)};
    }
  `
}

export const Container = styled.button<ContainerProps>`
  ${({ theme, size, fullWidth, hasIcon, minimal }) => css`
    display: inline-flex;
    align-items: center;
    justify-content: center;

    border: 0;
    padding: ${theme.spacings.xxsmall};
    border-radius: ${theme.border.radius};

    text-decoration: none;
    color: ${theme.colors.white};
    background: linear-gradient(180deg, #ff5f5f 0%, #f062c0 50%);

    &:hover {
      background: ${minimal
        ? 'none'
        : `linear-gradient(180deg, #e35565 0%, #d958a6 50%)`};
    }

    cursor: pointer;

    ${!!size && containerModifiers[size](theme)};
    ${!!fullWidth && containerModifiers.fullWidth()};
    ${!!hasIcon && containerModifiers.withIcon(theme)};
    ${!!minimal && containerModifiers.minimal(theme)};
  `}
`
