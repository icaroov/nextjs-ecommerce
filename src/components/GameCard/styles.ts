import styled, { css, DefaultTheme } from 'styled-components'
import media from 'styled-media-query'

export const Container = styled.article`
  ${({ theme }) => css`
    position: relative;

    display: flex;
    flex-direction: column;

    width: 100%;
    height: 100%;

    background-color: ${theme.colors.white};
  `}
`

export const ImageBox = styled.div`
  width: 100%;
  height: 14rem;

  background: #f6f7f8;
  background-size: 80rem 14rem;
  animation: placeholderShimmer 1s linear infinite forwards;
  background-image: linear-gradient(
    to right,
    #f6f7f8 0%,
    #edeef1 20%,
    #f6f7f8 40%,
    #f6f7f8 100%
  );

  img {
    width: 100%;
    height: 100%;
    object-fit: cover;
  }

  @keyframes placeholderShimmer {
    0% {
      background-position: -40rem 0;
    }
    100% {
      background-position: 40rem 0;
    }
  }
`

export const Content = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    position: relative;
    height: 100%;
    margin: ${theme.spacings.xsmall};
  `}
`

export const Info = styled.div`
  max-width: calc(100% - 2.5rem);
`

export const Title = styled.h3`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.medium};
    line-height: ${theme.font.sizes.medium};
    font-weight: ${theme.font.bold};
    color: ${theme.colors.black};
  `}
`

export const Developer = styled.h4`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.small};
    font-weight: ${theme.font.bold};
    color: ${theme.colors.gray};
  `}
`

export const FavButton = styled.div`
  ${({ theme }) => css`
    position: absolute;

    top: -0.5rem;
    right: 0;

    color: ${theme.colors.primary};
    cursor: pointer;

    svg {
      width: 2.5rem;
    }
  `}
`

export const BuyBox = styled.div`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    justify-content: flex-end;

    margin-top: ${theme.spacings.xxsmall};
  `}
`

type PriceProps = {
  isPromotional?: boolean
}

const priceModifiers = {
  default: (theme: DefaultTheme) => css`
    margin-right: calc(${theme.spacings.xxsmall} / 2);
    padding: 0 ${theme.spacings.xxsmall};

    border-radius: ${theme.border.radius};
    color: ${theme.colors.white};
    background-color: ${theme.colors.secondary};
  `,
  promotional: (theme: DefaultTheme) => css`
    margin-right: ${theme.spacings.xsmall};

    color: ${theme.colors.gray};
    text-decoration: line-through;
  `
}

export const Price = styled.div<PriceProps>`
  ${({ theme, isPromotional }) => css`
    display: inline-flex;
    align-items: center;
    height: 3rem;

    font-weight: ${theme.font.bold};

    ${media.lessThan('medium')`
      font-size: ${theme.font.sizes.small};
    `}

    ${!isPromotional && priceModifiers.default(theme)};
    ${isPromotional && priceModifiers.promotional(theme)};
  `}
`
