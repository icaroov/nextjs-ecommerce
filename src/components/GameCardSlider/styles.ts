import styled, { css } from 'styled-components'
import media from 'styled-media-query'

import { GameCardSliderProps } from '.'

type ContainerProps = Pick<GameCardSliderProps, 'color'>

export const Container = styled.section<ContainerProps>`
  ${({ theme, color = 'black' }) => css`
    ${media.lessThan('huge')`
      overflow: hidden;
    `}

    .slick-track,
    .slick-list {
      display: flex;
    }

    .slick-slide > div {
      flex: 1 0 auto;
      height: 100%;
      margin: 0 ${theme.spacings.xxsmall};
    }

    .slick-list {
      margin: 0 -${theme.spacings.xxsmall};
    }

    ${media.greaterThan('large')`
      .slick-slide > div {
        margin: 0 ${theme.spacings.xsmall};
      }

      .slick-list {
        margin: 0 -${theme.spacings.xsmall};
      }
    `}

    .slick-prev,
    .slick-next {
      position: absolute;

      display: block;
      padding: 0;
      top: 50%;

      width: 2.5rem;
      height: 2.5rem;

      transform: translate(0, -50%);

      color: ${theme.colors[color]};
      cursor: pointer;
    }

    .slick-prev {
      left: -${theme.spacings.xxlarge};
    }

    .slick-next {
      right: -${theme.spacings.xxlarge};
    }

    .slick-prev.slick-disabled,
    .slick-next.slick-disabled {
      visibility: hidden;
    }
  `}
`
