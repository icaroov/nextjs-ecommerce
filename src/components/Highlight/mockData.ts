export const HIGHLIGHT_MOCK = {
  title: 'Read Dead is back',
  subtitle: "Come see John's new adventures",
  backgroundImage: '/assets/img/red-dead-img.jpg',
  buttonLabel: 'Buy Now',
  buttonLink: '/games/rdr2'
}
