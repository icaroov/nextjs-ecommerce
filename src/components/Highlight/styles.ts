import styled, { css } from 'styled-components'
import media from 'styled-media-query'
import { HighlightProps } from '.'

type ContainerProps = Pick<HighlightProps, 'backgroundImage'>

export const Container = styled.section<ContainerProps>`
  ${({ backgroundImage }) => css`
    position: relative;

    display: grid;
    height: 23rem;

    background-image: url(${backgroundImage});
    background-position: center center;
    background-size: cover;

    &::after {
      content: '';
      position: absolute;

      width: 100%;
      height: 100%;

      background-color: rgba(0, 0, 0, 0.6);
    }

    ${media.greaterThan('medium')`
      height: 32rem;
    `}
  `}
`

export const Content = styled.div`
  ${({ theme }) => css`
    z-index: ${theme.layers.base};
    text-align: end;
    padding: ${theme.spacings.xsmall};

    ${media.greaterThan('medium')`
      padding: ${theme.spacings.large};
      align-self: end;
    `}
  `}
`

export const Title = styled.h2`
  ${({ theme }) => css`
    font-weight: ${theme.font.bold};
    color: ${theme.colors.white};

    ${media.greaterThan('medium')`
      font-size: ${theme.font.sizes.xxlarge};
    `}
  `}
`

export const Subtitle = styled.h3`
  ${({ theme }) => css`
    font-weight: ${theme.font.light};
    color: ${theme.colors.white};
    margin-bottom: ${theme.spacings.medium};

    ${media.greaterThan('medium')`
      font-size: ${theme.font.sizes.large};
    `}
  `}
`
