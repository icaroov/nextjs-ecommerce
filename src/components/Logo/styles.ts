import styled, { css } from 'styled-components'

import { LogoProps } from '.'

export const Container = styled.div<LogoProps>`
  ${({ theme, color = 'white' }) => css`
    color: ${theme.colors[color]};
  `}
`
