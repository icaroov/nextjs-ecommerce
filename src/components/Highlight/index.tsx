import Button from 'components/Button'
import * as Styled from './styles'

export type HighlightProps = {
  title: string
  subtitle: string
  buttonLabel: string
  buttonLink: string
  backgroundImage: string
  floatImage?: string
  alignLeft?: boolean
}

const Highlight = ({
  title,
  subtitle,
  buttonLabel,
  buttonLink,
  backgroundImage,
  floatImage,
  alignLeft = false
}: HighlightProps) => (
  <Styled.Container backgroundImage={backgroundImage} alignLeft={alignLeft}>
    {!!floatImage && <Styled.FloatImage src={floatImage} alt={title} />}
    <Styled.Content>
      <Styled.Title>{title}</Styled.Title>
      <Styled.Subtitle>{subtitle}</Styled.Subtitle>
      <Button as="a" href={buttonLink}>
        {buttonLabel}
      </Button>
    </Styled.Content>
  </Styled.Container>
)

export default Highlight
