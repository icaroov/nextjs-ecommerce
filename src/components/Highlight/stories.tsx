import { Story, Meta } from '@storybook/react/types-6-0'

import Highlight, { HighlightProps } from '.'

import HIGHLIGHT_MOCK from './mockData'

export default {
  title: 'Highlight',
  component: Highlight,
  args: { ...HIGHLIGHT_MOCK }
} as Meta

export const Default: Story<HighlightProps> = args => (
  <div style={{ maxWidth: '104rem', margin: '0 auto' }}>
    <Highlight {...args} />
  </div>
)

export const WithFloatImage: Story<HighlightProps> = args => (
  <div style={{ maxWidth: '104rem', margin: '0 auto' }}>
    <Highlight {...args} />
  </div>
)

WithFloatImage.args = {
  floatImage: '/assets/img/red-dead-float.png'
}
