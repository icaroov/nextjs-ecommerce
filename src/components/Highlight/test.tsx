import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import Highlight from '.'
import * as Styled from './styles'

const props = {
  title: 'Heading 1',
  subtitle: 'Heading 2',
  backgroundImage: '/assets/img/red-dead-img.jpg',
  buttonLabel: 'Buy Now',
  buttonLink: '#'
}

describe('<Highlight />', () => {
  it('should render the heading', () => {
    const { container } = renderWithTheme(<Highlight {...props} />)

    expect(
      screen.getByRole('heading', { name: /heading 1/i })
    ).toBeInTheDocument()

    expect(
      screen.getByRole('heading', { name: /heading 2/i })
    ).toBeInTheDocument()

    expect(screen.getByRole('link', { name: /buy now/i })).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the background image', () => {
    const { container } = renderWithTheme(<Highlight {...props} />)

    expect(container.firstChild).toHaveStyle({
      backgroundImage: `url(${props.backgroundImage})`
    })
  })

  it('should render the float image image', () => {
    renderWithTheme(<Highlight {...props} floatImage="float-image.png" />)

    expect(screen.getByRole('img', { name: props.title })).toHaveAttribute(
      'src',
      'float-image.png'
    )
  })

  it('should render align right by default', () => {
    const { container } = renderWithTheme(<Highlight {...props} />)

    expect(container.firstChild).toHaveStyleRule(
      'grid-template-areas',
      "'floatimage content'"
    )
  })

  it('should render align left', () => {
    const { container } = renderWithTheme(<Highlight {...props} alignLeft />)

    expect(container.firstChild).toHaveStyleRule(
      'grid-template-areas',
      "'content floatimage'"
    )

    expect(container.firstChild).toHaveStyleRule('text-align', 'left', {
      modifier: `${Styled.Content}`
    })

    expect(container.firstChild).toHaveStyleRule('justify-self', 'end', {
      modifier: `${Styled.FloatImage}`
    })
  })
})
