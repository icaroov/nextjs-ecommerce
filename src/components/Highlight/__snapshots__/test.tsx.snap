// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Highlight /> should render the heading 1`] = `
.c4 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border: 0;
  padding: 0.8rem;
  border-radius: 0.4rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #FAFAFA;
  background: linear-gradient(180deg,#ff5f5f 0%,#f062c0 50%);
  cursor: pointer;
  height: 4rem;
  font-size: 1.4rem;
  padding: 0.8rem 3.2rem;
}

.c4:hover {
  background: linear-gradient(180deg,#e35565 0%,#d958a6 50%);
}

.c0 {
  position: relative;
  display: grid;
  grid-template-areas: 'floatimage content';
  grid-template-columns: 1.3fr 2fr;
  height: 23rem;
  background-image: url(/assets/img/red-dead-img.jpg);
  background-position: center center;
  background-size: cover;
}

.c0::after {
  content: '';
  position: absolute;
  width: 100%;
  height: 100%;
  background-color: rgba(0,0,0,0.6);
}

.c1 {
  grid-area: content;
  z-index: 10;
  text-align: right;
  padding: 1.6rem;
}

.c2 {
  font-size: 1.8rem;
  font-weight: 600;
  color: #FAFAFA;
}

.c3 {
  margin-bottom: 3.2rem;
  font-size: 1.4rem;
  font-weight: 300;
  color: #FAFAFA;
}

@media (min-width:768px) {
  .c0 {
    height: 32rem;
  }
}

@media (min-width:768px) {
  .c1 {
    padding: 4.0rem;
    -webkit-align-self: end;
    -ms-flex-item-align: end;
    align-self: end;
  }
}

@media (min-width:768px) {
  .c2 {
    font-size: 2.8rem;
  }
}

@media (min-width:768px) {
  .c3 {
    font-size: 1.8rem;
  }
}

<section
  class="c0"
>
  <div
    class="c1"
  >
    <h2
      class="c2"
    >
      Read Dead is back
    </h2>
    <h3
      class="c3"
    >
      Come see John's new adventures
    </h3>
    <a
      class="c4"
      href="/games/rdr2"
    >
      <span>
        Buy Now
      </span>
    </a>
  </div>
</section>
`;
